File: NetBeansProjects/WebApplication1/src/java/com/epic/canvascontrollibrary/CCLCalenderProps.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.epic.canvascontrollibrary;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Gilgamesh
 */
    public class CCLCalenderProps
    {
        public String CanvasID;
        public String WindowID;
        public String X;
        public String Y;
        public String Width;
        public String Height;
        public String VisibleMonth;
        public String VisibleYear;
        public String SelectedDay;
        public String DayCellWidth;
        public String DayCellHeight;
        public String HeaderHeight;
        public String TextHeaderColor;
        public String TextHeaderHeight;
        public String TextHeaderFontString;
        public String DayDateActiveColor;
        public String DayDateActiveTextHeight;
        public String DayDateActiveTextFontString;
        public String DayDateInactiveTextColor;
        public String DayDateInactiveTextHeight;
        public String DayDateInactiveTextFontString;
        public String SelectedDayTextColor;
        public String SelectedDayTextHeight;
        public String SelectedDayTextFontString;
        public String SelectedDayHighLightColor;
        public String TodayTextColor;
        public String TodayTextHeight;
        public String TodayTextFontString;
        public String TodayHighLightColor;
        public String OnDayClickFunction;
        public String HeaderBackgroundColor;
        public String BodyBackgroundColor;
        public String MouseOverHightLightColor;
        public String MouseHoverDate;
        public List<Object> ButtonClickExtents;
        public List<Object> DateClickExtents;
        public String DayLabelTextColor;
        public String DayLabelTextHeight;
        public String DayLabelTextFontString;
        public Object Tag;
        public String TabStopIndex;

        CCLCalenderProps()
        {
            ButtonClickExtents = new ArrayList<Object>();
            DateClickExtents = new ArrayList<Object>();
        }
    }

File: ASP.NET MVC Web Api New Solution/CanvasControlLibrary/CanvasControlLibrary/Models/CCLCalenderProps.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations.Schema;

namespace CanvasControlLibrary.Models
{
    public class CCLCalenderProps
    {
        public class DateClickExtent
        {
            public int X { get; set; }
            public int Y { get; set; }
            public DateTime Date { get; set; }
        }

        public class ButtonClickExtent
        {
            public int X { get; set; }
            public int Y { get; set; }
            public int Width { get; set; }
            public int Height { get; set; }
        }

        public string CanvasID { get; set; }
        public int WindowID { get; set; }
        public int X { get; set; }
        public int Y { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }
        public string VisibleMonth { get; set; }
        public string VisibleYear { get; set; }
        public string SelectedDay { get; set; }
        public int DayCellWidth { get; set; }
        public int DayCellHeight { get; set; }
        public int HeaderHeight { get; set; }
        public string TextHeaderColor { get; set; }
        public int TextHeaderHeight { get; set; }
        public string TextHeaderFontString { get; set; }
        public string DayDateActiveColor { get; set; }
        public int DayDateActiveTextHeight { get; set; }
        public string DayDateActiveTextFontString { get; set; }
        public string DayDateInactiveTextColor { get; set; }
        public int DayDateInactiveTextHeight { get; set; }
        public string DayDateInactiveTextFontString { get; set; }
        public string SelectedDayTextColor { get; set; }
        public int SelectedDayTextHeight { get; set; }
        public string SelectedDayTextFontString { get; set; }
        public string SelectedDayHighLightColor { get; set; }
        public string TodayTextColor { get; set; }
        public int TodayTextHeight { get; set; }
        public string TodayTextFontString { get; set; }
        public string TodayHighLightColor { get; set; }
        public string HeaderBackgroundColor { get; set; }
        public string BodyBackgroundColor { get; set; }
        public string MouseOverHightLightColor { get; set; }
        public string MouseHoverDate { get; set; }
        public List<ButtonClickExtent> ButtonClickExtents { get; set; }
        public List<DateClickExtent> DateClickExtents { get; set; }
        public string DayLabelTextColor { get; set; }
        public int DayLabelTextHeight { get; set; }
        public string DayLabelTextFontString { get; set; }
        public object Tag { get; set; }
        public int TabStopIndex { get; set; }

        public CCLCalenderProps()
        {
            ButtonClickExtents = new List<ButtonClickExtent>();
            DateClickExtents = new List<DateClickExtent>();
        }
    }
}